src/components/MapArea.js

```diff
@@ -40,84 +50,86 @@ const trailData = {
     "desc": "Generated by AllTrails https://www.alltrails.com/"
   }
-}
+};
 
 
 export default function MapArea() {
-
-  const [ style, setStyle ] = useState("mapbox://styles/mapbox/outdoors-v11" );
-  const [ latitude, setLatitude ] = useState(45.393);
-  const [ longitude, setLongitude ]= useState(-75.511);
-  const [ zoom, setZoom ] = useState(15);
-  const mapContainer = useRef(null)
-	const map = useRef(null);
-	// this is where all of our map logic is going to live  
-	// adding the empty dependency array ensures that the map  
-	// is only created once  
-	useEffect(() => {    
-		// create the map and configure it    
-		// check out the API reference for more options    
-		map.current = new mapboxgl.Map({      
-			container: mapContainer.current,      
-			style,      
-			center: [longitude, latitude],      
-			zoom,    
-		})
-
-    map.current.on('load', function() {
-      map.current.addSource('trailData', {
-        'type': 'geojson',
-        'data': trailData
-      })
-  
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+
+  const displayDataOnMap = (map, trailData) => {
+    if (!map.current.getSource("trailData")) {
+      map.current.addSource("trailData", {
+        type: "geojson",
+        data: trailData,
+      });
+    }
+
+    if (!map.current.getLayer("trailData")) {
       map.current.addLayer({
-        'id': 'trailData',
-        'type': 'line',
-        'source': 'trailData',
-        'layout': {},
-        'paint': {
-          'line-color': '#ec2222',
-          'line-width': 2,
-          'line-dasharray': [2, 1]
-        }
-      })
-    })
-
-		return () => map.current.remove() 
-
-	}, [style])
-
+        id: "trailData",
+        type: "line",
+        source: "trailData",
+        layout: {},
+        paint: {
+          "line-color": "#ec2222",
+          "line-width": 2,
+          "line-dasharray": [2, 1],
+        },
+      });
+    }
+  };
 
   useEffect(() => {
-    if (!map.current) return; // wait for map to initialize
-    map.current.on('move', () => {
-      setLongitude(map.current.getCenter().lng.toFixed(4));
-      setLatitude(map.current.getCenter().lat.toFixed(4));
-      setZoom(map.current.getZoom().toFixed(2));
+    // Render a map and configure it
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: "mapbox://styles/mapbox/outdoors-v11",
+      ...DEFAULT_VIEWPORT,
     });
-  });
+
+    map.current.on("load", () => {
+      displayDataOnMap(map, trailData);
+    });
+
+    return () => map.current.remove();
+  }, []);
 
   const changeStyle = (styleURL) => {
-    setStyle(styleURL);
-  }
+    if (map.current.getLayer("trailData")) {
+      map.current.removeLayer("trailData");
+    }
+    if (map.current.getSource("trailData")) {
+      map.current.removeSource("trailData");
+    }
+    map.current.setStyle(styleURL);
+    map.current.on("style.load", () => {
+      displayDataOnMap(map, trailData);
+    });
```

Comment:
since this is a map event hook, it should be defined when we initialize the map.

Reply:
displayDataOnMap is also called once when the map is initialized. 106-108 is for when the baselayer style is changed. Line 105 calls map.current.setStyle(styleURL) which, from what I've read, takes some time to change the style. 106-108 will wait for the new style to finish applying before calling displayDataOnMap to reapply the source and layers. Without 106 and 108 then the data may not show, or errors such as "There is no source with this ID" when trying to remove the source prior to calling map.current.setStyle(styleURL)

Reply:
It should work if you move the `style.load` event hook to the use effect where we define a new map.  We probably won't nee the `load` event hook anymore if `style.load` is being called when the baselayer is initially being set.